### ISSUE TITLE
Invocation of LC.exe in the LC Task

### ISSUE THREAD

**Issue by LambdaSix** [Microsoft/msbuild#914]
https://github.com/Microsoft/msbuild/blob/master/src/XMakeTasks/LC.cs#L184 states that when building against a framework > 4.0, lc.exe will accept an @response.rsp file.

This actually only works for the lc.exe in the v10.0A SDK for the 4.6.2 Tools. Seems like the version check should be against 4.6.2 instead of 4.0?

**Comment by bbaumann** [comment]
I confirm this bug.
I have spent the day trying to debug a license not being generated on the build server.

* The command line logged worked when typed directly in the cli.
* MSBuild invokation of lc.exe fails with the logo (the arguments seems not to be understood)
* Further investigation show that msbuild launch lc.exe with a rsp temp file
* copying this rsp temp file and launching lc as msbuild does reproduce the issue

I am using .NET 4.5.1 toolkit (sdk v8.0A).

**Comment by rainersigwald** [comment]
Did some source-history spelunking today. `lc.exe` was taught to handle a response file in July 2014, and the task was changed to use response files unconditionally in January 2015. Later that month it was changed to the current form, using 4.0 as a cutoff.

On my current machine, I have many different `lc.exe`s:
```
C:\Program Files (x86)\Microsoft SDKs>dir /s /b lc.exe
C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.6 Tools\lc.exe
C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.6 Tools\x64\lc.exe
C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.6.1 Tools\lc.exe
C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.6.1 Tools\x64\lc.exe
C:\Program Files (x86)\Microsoft SDKs\Windows\v7.0A\Bin\lc.exe
C:\Program Files (x86)\Microsoft SDKs\Windows\v7.0A\Bin\x64\lc.exe
C:\Program Files (x86)\Microsoft SDKs\Windows\v8.1A\bin\NETFX 4.5.1 Tools\lc.exe
C:\Program Files (x86)\Microsoft SDKs\Windows\v8.1A\bin\NETFX 4.5.1 Tools\x64\lc.exe
```

It looks like the real cutoff is at .NET 4.6:
```
C:\>"C:\Program Files (x86)\Microsoft SDKs\Windows\v8.1A\bin\NETFX 4.5.1 Tools\lc.exe" /?
Microsoft (R) .NET License Compiler
[Microsoft .Net Framework, Version 4.0.30319.42000]
Copyright (C) Microsoft Corporation.  All rights reserved.


Generates a .NET Licenses file and adds it to the manifest of the given assembly
Usage:
    lc /target:TargetAssembly /complist:filename [/outdir:path] [/i:modules] [/v] [/nologo]

Options:
    /target:<str>              Target assembly for the generated licenses file
    /complist:<str>            Licensed component list file
    /outdir:<str>              Output directory for the generated licenses file
    /i:<str>                   Specify modules to load
    /v                         Verbose output
    /nologo                    Suppress the display of the startup banner


C:\>"C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.6 Tools\lc.exe" /?
Microsoft (R) .NET License Compiler
[Microsoft .Net Framework, Version 4.6.81.0]
Copyright (C) Microsoft Corporation.  All rights reserved.


Generates a .NET Licenses file and adds it to the manifest of the given assembly
Usage:
    lc /target:TargetAssembly /complist:filename [/outdir:path] [/i:modules] [/v] [/nologo]

Options:
    /target:<str>              Target assembly for the generated licenses file
    /complist:<str>            Licensed component list file
    /outdir:<str>              Output directory for the generated licenses file
    /i:<str>                   Specify modules to load
    /v                         Verbose output
    /nologo                    Suppress the display of the startup banner
    @<file>                    Accept options from a response file
```

The targets use `lc.exe` from [the target sdktools directory](https://github.com/Microsoft/msbuild/blob/a66f1cdcfca7910f9460f44343030f238b632258/src/XMakeTasks/Microsoft.Common.CurrentVersion.targets#L2928), so I think the right cutoff is `4.6`.

**Comment by bbaumann** [comment]
Thank you Rainer, I did not have the energy to install the sdk and locate the real cutoff. I updated my PR to reflect your finding.

**Comment by rainersigwald** [comment]
This was fixed a loooooong time ago by @bbaumann in #1414. Thanks again!
